src/common/global-event.tsx

```diff
@@ -0,0 +1,44 @@
+import { Component } from 'react';
+
+export type Props = {
+    [EventName in keyof WindowEventMap]?: (event: WindowEventMap[EventName]) => void;
+};
+
+export default class GlobalEvent extends Component<Props> {
+
+    public componentDidMount() {
+        this.forEachEvent((name, listener) => this.subscribe(name, listener));
+    }
+
+    public componentWillUnmount() {
+        this.forEachEvent((name, listener) => this.unsubscribe(name, listener));
+    }
+
+    public componentWillReceiveProps(props: Props) {
+        this.forEachEvent((name, listener) => {
+            if (listener !== props[name]) {
+                this.unsubscribe(name, listener);
+                props[name] && this.subscribe(name, props[name] as EventListener);
+            }
+        });
+    }
+
+    public render() {
+        return null;
+    }
+
+    private forEachEvent(fn: (name: keyof Props, listener: EventListener) => void) {
+        (Object
+            .keys(this.props)
+            .filter(name => name !== 'children') as Array<keyof Props>)
+            .forEach(name => fn(name, this.props[name] as EventListener));
+    }
+
+    private subscribe(event: string, listener: EventListener) {
+        window && window.addEventListener(event, listener);
```

Comment:
don't you need to check something like `typeof window !== "undefined"`? is this tested in node?

Reply:
@idoros 
Yes but indirectly, tbh. When this component is used in some other component that is being tested for SSR. Should we add `<GlobalEvent />` one to `src/index.ts` reexports, too? This way it will get tested for SSR.